fh-js-sdk.d.ts

```diff
@@ -0,0 +1,503 @@
+// Type definitions for fh-js-sdk 2.17.5
+// Project: fh-js-sdk
+// Definitions by: Aiden Keating <akeating@redhat.com>
+
+/** @module fh-js-sdk */
+declare module 'fh-js-sdk' {
+
+    /**
+     * Interface for the object with the metadata added by the FH SDK for each cloud request.
+     * 
+     * @type FHParams
+     */
+    export interface FHParams {
+        appid?:string;
+        appkey?:string;
+        projectid?:string;
+        cuid?:string;
+        destination?:string;
+        sdk_version?:string;
+        connectiontag?:string;
+    }
+
+    /**
+     * Interface for the data provided within a PushNotificationCallback argument.
+     * 
+     * @type PushNotificationData
+     */
+    export interface PushNotificationData {
+        alert:string;
+        sound:string;
+        badge:string;
+        coldstart:boolean;
+        payload:{};
+    }
+
+    /**
+     * Interface for the configuration object provided to the push command.
+     * 
+     * @type PushConfig
+     */
+    export interface PushConfig {
+        alias:string;
+        categories:Array<string>;
+    }
+
+    /**
+     * Interface for the default error object used with the error callback in the SDK.
+     * 
+     * @type FailureCallbackError
+     */
+    export interface FailureCallbackError {
```

Comment:
We'd be best to use true Error instances for all errors. This has been a pain point for a while and would be best addressed before merging this PR so as to finally fix this issue.

Thoughts